auto_nag/scripts/inactive_reviewer.py

```diff
@@ -0,0 +1,233 @@
+
+        rev_ids = {rev_id for bug in bugs.values() for rev_id in bug["rev_ids"]}
+        revisions = self._get_revisions_with_inactive_reviewers(list(rev_ids))
+
+        for bugid, bug in list(bugs.items()):
+            inactive_revs = [
+                revisions[rev_id] for rev_id in bug["rev_ids"] if rev_id in revisions
+            ]
+            if inactive_revs:
+                bug["revisions"] = inactive_revs
+                self._add_needinfo(bugid, inactive_revs)
+            else:
+                del bugs[bugid]
+
+        # Resolving https://github.com/mozilla/relman-auto-nag/issues/1300 should clean this
+        # including improve the wording in the template (i.e., "See the search query on Bugzilla").
+        self.query_url = utils.get_bz_search_url({"bug_id": ",".join(bugs.keys())})
+
+        return bugs
+
+    def _add_needinfo(self, bugid: str, inactive_revs: list) -> None:
+        ni_mails = {rev["author"]["name"] for rev in inactive_revs}
+        nicknames = utils.english_list(
+            sorted({rev["author"]["nick"] for rev in inactive_revs})
+        )
+        comment = self.ni_template.render(
+            revisions=inactive_revs,
+            nicknames=nicknames,
+            plural=utils.plural,
+            documentation=self.get_documentation(),
+        )
+
+        self.autofix_changes[bugid] = {
+            "comment": {"body": comment},
+            "flags": [
+                {
+                    "name": "needinfo",
+                    "requestee": ni_mail,
+                    "status": "?",
+                    "new": "true",
+                }
+                for ni_mail in ni_mails
+            ],
+        }
+
+    def _get_revisions_with_inactive_reviewers(self, rev_ids: list) -> Dict[int, dict]:
+        revisions: List[dict] = []
+        for _rev_ids in Connection.chunks(rev_ids, PHAB_CHUNK_SIZE):
+            for revision in self._fetch_revisions(_rev_ids):
+                if (
+                    len(revision["attachments"]["reviewers"]["reviewers"]) == 0
+                    or revision["fields"]["status"]["value"] != "needs-review"
+                    or revision["fields"]["isDraft"]
+                ):
+                    continue
+
+                reviewers = [
+                    {
+                        "phid": reviewer["reviewerPHID"],
+                        "is_group": reviewer["reviewerPHID"].startswith("PHID-PROJ"),
+                        "is_blocking": reviewer["isBlocking"],
+                        "is_accepted": reviewer["status"] == "accepted",
+                    }
+                    for reviewer in revision["attachments"]["reviewers"]["reviewers"]
+                ]
+
+                # Group reviewers will be consider always active; so if there is
+                # no other reviewers blocking, we don't need to check further.
+                if any(
+                    reviewer["is_group"] or reviewer["is_accepted"]
```

Comment:
We are dropping the whole revision using `continue`:

https://github.com/mozilla/relman-auto-nag/blob/2381415ed42035a82a919f43e9ff894f8988b37e/auto_nag/scripts/inactive_reviewer.py#L113-L123

If you mean not including these reviewers in the first place, we need to know the full list of reviewers before deciding on a revision.

Now we are filtering the reviewers in one place only:

https://github.com/mozilla/relman-auto-nag/blob/2381415ed42035a82a919f43e9ff894f8988b37e/auto_nag/scripts/inactive_reviewer.py#L145-L152

Reply:
I was thinking of removing e.g. the "group" reviewers from the reviewer list here, so later we don't need to filter them out.

(not important, you can also keep it as it is)

Reply:
I keep them because we consider group reviewers as active reviewers; thus, if there are no blocking reviewers who are inactive, we do not warn about the patch (we do not warn if we have at least one active reviewer unless the inactive reviewer is blocking). If I remove group reviewers from the initial list, we cannot tell if the revision has an active reviewer.